### PR TITLE
Add named exports

### PR DESCRIPTION
**Pull request by Richienb**
- Export `chalk.Instance` as `Chalk` instead
- Export `chalk.supportsColor` as `supportsColor` instead
- Export `chalk.stderr` as `chalkStderr` instead
- Export `chalk.stderr.supportsColor` as `supportsColorStderr` instead

#431

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -149,11 +149,11 @@ interface ChalkFunction {
 	(...text: unknown[]): string;
 }
 
-export interface Chalk extends ChalkFunction {
+export interface ChalkInstance extends ChalkFunction {
 	/**
 	Return a new Chalk instance.
 	*/
-	Instance: ChalkInstance;
+	Chalk: ChalkClass;
```
Comment (Richienb) [inline index.d.ts:156]: Should this still be `Instance` instead of `Chalk`

Comment (sindresorhus) [reply index.d.ts:156]: It should be removed here. It's already exported as a named export.

File: index.d.ts [hunk 2]
```diff
@@ -194,40 +194,40 @@ export interface Chalk extends ChalkFunction {
...
+	hwb: (hue: number, whiteness: number, blackness: number) => ChalkInstance;
 
 	/**
 	Use a [Select/Set Graphic Rendition](https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters) (SGR) [color code number](https://en.wikipedia.org/wiki/ANSI_escape_code#3/4_bit) to set text color.
 
 	30 <= code && code < 38 || 90 <= code && code < 98
 	For example, 31 for red, 91 for redBright.
 	*/
-	ansi: (code: number) => Chalk;
+	ansi: (code: number) => ChalkInstance;
```
Comment (sindresorhus) [inline index.d.ts:225]: Couldn't this return `this`?

File: index.d.ts [hunk 3]
```diff
@@ -403,9 +403,16 @@ Order doesn't matter, and later styles take precedent in case of a conflict.
...
 This simply means that `chalk.red.yellow.green` is equivalent to `chalk.green`.
 */
-declare const chalk: Chalk & ChalkFunction & {
-	supportsColor: ColorSupport | false;
-	stderr: Chalk & {supportsColor: ColorSupport | false};
+declare const chalk: ChalkInstance & ChalkFunction;
+
+declare const supportsColor: ColorSupport | false;
+declare const chalkStderr: ChalkInstance & {supportsColor: ColorSupport | false};
+declare const Chalk: ChalkClass;
```
Comment (sindresorhus) [inline index.d.ts:410]: You can just export these directly. No need for `declare`.

File: index.d.ts [hunk 4]
```diff
@@ -403,9 +403,16 @@ Order doesn't matter, and later styles take precedent in case of a conflict.
 
 This simply means that `chalk.red.yellow.green` is equivalent to `chalk.green`.
 */
-declare const chalk: Chalk & ChalkFunction & {
-	supportsColor: ColorSupport | false;
-	stderr: Chalk & {supportsColor: ColorSupport | false};
+declare const chalk: ChalkInstance & ChalkFunction;
+
+declare const supportsColor: ColorSupport | false;
+declare const chalkStderr: ChalkInstance & {supportsColor: ColorSupport | false};
```
Comment (sindresorhus) [inline index.d.ts:409]: `{supportsColor: ColorSupport | false}` should be removed.

Comment (sindresorhus) [reply index.d.ts:409]: I think it could just be `export const chalkStderr: typeof chalk;`

File: index.d.ts [hunk 5]
```diff
@@ -93,7 +93,7 @@ export interface Options {
 /**
 Return a new Chalk instance.
 */
-export type ChalkInstance = new (options?: Options) => Chalk;
+export type ChalkClass = new (options?: Options) => ChalkInstance;
```
Comment (sindresorhus) [inline index.d.ts:96]:
```suggestion
export type Chalk = new (options?: Options) => ChalkInstance;
```

?

File: index.d.ts [hunk 6]
```diff
@@ -403,9 +398,9 @@ Order doesn't matter, and later styles take precedent in case of a conflict.
...
 This simply means that `chalk.red.yellow.green` is equivalent to `chalk.green`.
 */
-declare const chalk: Chalk & ChalkFunction & {
-	supportsColor: ColorSupport | false;
-	stderr: Chalk & {supportsColor: ColorSupport | false};
-};
+declare const chalk: ChalkInstance & ChalkFunction;
+
+export const supportsColor: ColorSupport | false;
+export const chalkStderr: typeof chalk & {supportsColor: typeof supportsColor};
```
Comment (sindresorhus) [inline index.d.ts:404]: Should we maybe export a `supportsColorStderr` instead? For consistency.

**Comment by sindresorhus** [comment]
I don't think https://github.com/chalk/chalk/pull/432/files#r615220658 was solved. You need to remove the `& ..`.

**Pull request merged**